Contest_Package.md

```diff
@@ -40,37 +40,35 @@ Information in the API is always either in JSON format,
 - The NDJSON returned from the endpoint `/contests/<id>/<endpoint>` is stored as
   `<endpoint>.ndjson`. (The only such endpoint is `event-feed`.)
 
-
-Files referenced in `api.json` and `contest.json` are stored as `api/<filename>`
-and `contest/<filename>` respectively, and files referenced in
-`<endpoint>.json` are stored as `<endpoint>/<id>/<filename>`, where:
-
-- `<id>` is the ID of the endpoint object the reference is in.
-- `<filename>` is the filename specified in the file reference object.
-
 Some of these API endpoints are often written by humans in a package. For this
-reason, those files can also be written in YAML instead of JSON. This holds for
-the following files:
+reason, those files can also be written in YAML instead of JSON in a Contest
+Package. This holds for the following files:
 
 - `contest.json`, which then becomes `contest.yaml`.
 - `problems.json`, which then becomes `problems.yaml`.
 - `accounts.json`, which then becomes `accounts.yaml`.
 
 The section [Example YAML files](#example-yaml-files) list example YAML files.
 
+### File references
+
+Files referenced in `api.json` and `contest.json` are stored as `api/<filename>`
+and `contest/<filename>` respectively, and files referenced in
+`<endpoint>.json` are stored as `<endpoint>/<id>/<filename>`, where:
+
+- `<id>` is the ID of the endpoint object the reference is in.
+- `<filename>` is the filename specified in the file reference object.
+
 Note that the API specification requires that filenames are unique within
 endpoint objects, so this is always possible.
 
-It is not required that a URL specified in an href is always valid.
-Specifically, in many cases the contest is running in a local network that is
-taken down after the contest, and in this case the URL would definitely not
-still be working. To keep the archiving process as simple as possible, stale
-URLs do not have to be removed from the data, but due to this the URLs should
-be ignored and the file stored as specified above should be used instead.
-
-Optionally one could create a Shallow Package by not storing the files, in which
-case the URLs must be valid. This could be useful in some cases where the size
-of the package matters.
+The file reference may optionally contain a URL in the `href`
+property, for example when the package is archived from a Contest API.
+However, if a file is present in the package at the expected storage
+location, then that should be used and the original URL ignored. If no
+such file exists, then the original URL should be used to retrieve the
+file. This option could be used to create a shallow package, when the
+size of the package matters.
```

Comment:
I think this is *less* clear than the old version.

Reply:
Rewritten, keeping most of the old text to explain the why.

Reply:
I think this change makes you more happy than it makes me unhappy, so in total happiness increases. I'm fine with this.